tests/calico_cni_k8s_test.go

```diff
@@ -52,6 +54,25 @@ func ensureNamespace(clientset *kubernetes.Clientset, name string) {
 	Expect(err).NotTo(HaveOccurred())
 }
 
+func ensurePodDeleted(clientset *kubernetes.Clientset, ns string, podName string) {
+	err := clientset.CoreV1().Pods(ns).Delete(podName, metav1.NewDeleteOptions(0))
+	if kerrors.IsNotFound(err) {
+		return
+	}
+	if err != nil {
+		panic(err)
+	}
+	Eventually(func() bool {
+		pod, err := clientset.CoreV1().Pods(ns).Get(podName, metav1.GetOptions{})
+		if kerrors.IsNotFound(err) {
+			return true
+		}
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
Hm, I'm not sure if you can do an Expect inside an Eventually, I haven't seen that before.

Usually for this pattern we use a `func() error` and then expect it to be nil, returning an error whenever we haven't met the desired criteria.